plugin/tower/pkg/controller/endpoint/controller.go

```diff
@@ -245,20 +298,137 @@ func (c *Controller) enqueueVMNics(vm *schema.VM) {
 
 func (c *Controller) addEndpoint(new interface{}) {
 	obj := new.(*v1alpha1.Endpoint)
-	c.endpointQueue.Add(obj.GetName())
+	if strings.HasPrefix(obj.GetName(), StaticEndpointPrefix) {
+		c.staticEndpointQueue.Add(obj.GetName())
+	} else {
+		c.endpointQueue.Add(obj.GetName())
+	}
 }
 
 func (c *Controller) updateEndpoint(old interface{}, new interface{}) {
-	newEp := new.(*v1alpha1.Endpoint)
-	c.endpointQueue.Add(newEp.GetName())
+	obj := new.(*v1alpha1.Endpoint)
+	if strings.HasPrefix(obj.GetName(), StaticEndpointPrefix) {
+		c.staticEndpointQueue.Add(obj.GetName())
+	} else {
+		c.endpointQueue.Add(obj.GetName())
+	}
 }
 
 func (c *Controller) deleteEndpoint(old interface{}) {
 	if d, ok := old.(cache.DeletedFinalStateUnknown); ok {
 		old = d.Obj
 	}
 	obj := old.(*v1alpha1.Endpoint)
-	c.endpointQueue.Add(obj.GetName())
+	if strings.HasPrefix(obj.GetName(), StaticEndpointPrefix) {
+		c.staticEndpointQueue.Add(obj.GetName())
+	} else {
+		c.endpointQueue.Add(obj.GetName())
+	}
+}
+
+func (c *Controller) addEverouteCluster(new interface{}) {
+	cluster := new.(*schema.EverouteCluster)
+	for _, controller := range cluster.ControllerInstances {
+		c.staticEndpointQueue.Add(c.getCtrlEndpointName(cluster.ID, controller))
+	}
+}
+
+func (c *Controller) deleteEverouteCluster(old interface{}) {
+	if d, ok := old.(cache.DeletedFinalStateUnknown); ok {
+		old = d.Obj
+	}
+	cluster := old.(*schema.EverouteCluster)
+	for _, ctrl := range cluster.ControllerInstances {
+		if ctrl.IPAddr != "" {
```

Comment:
Who will response for the IP address validation?

Reply:
No one will not valid it.
add IP valid on ADD and UPDATE op.